src/pykeen/models/nbase.py

```diff
@@ -0,0 +1,475 @@
+
+
+def _prepare_representation_module_list(
+    representations: EmbeddingSpecificationHint,
+    num_embeddings: int,
+    shapes: Sequence[str],
+    label: str,
+    skip_checks: bool = False,
+) -> Sequence[NewRepresentationModule]:
+    """Normalize list of representations and wrap into nn.ModuleList."""
+    # Important: use ModuleList to ensure that Pytorch correctly handles their devices and parameters
+    if representations is None:
+        representations = []
+    if not isinstance(representations, Sequence):
+        representations = [representations]
+    if not skip_checks and len(representations) != len(shapes):
+        raise ValueError(
+            f"Interaction function requires {len(shapes)} {label} representations, but "
+            f"{len(representations)} were given.",
+        )
+    modules = []
+    for r in representations:
+        if not isinstance(r, NewRepresentationModule):
+            assert isinstance(r, EmbeddingSpecification)
+            r = r.make(num_embeddings=num_embeddings)
+        if r.max_id < num_embeddings:
+            raise ValueError(
+                f"{r} only provides {r.max_id} {label} representations, but should provide {num_embeddings}.",
+            )
+        elif r.max_id > num_embeddings:
+            logger.warning(
+                f"{r} provides {r.max_id} {label} representations, although only {num_embeddings} are needed."
+                f"While this is not necessarily wrong, it can indicate an error where the number of {label} "
+                f"representations was chosen wrong.",
+            )
+        modules.append(r)
+    if not skip_checks:
+        check_shapes(*zip(
+            (r.shape for r in modules),
+            shapes,
+        ), raise_on_errors=True)
+    return nn.ModuleList(modules)
+
+
+class ERModel(
+    Generic[HeadRepresentation, RelationRepresentation, TailRepresentation],
+    _NewAbstractModel,
+    autoreset=False,
+):
+    """A commonly useful base for KGEMs using embeddings and interaction modules."""
+
+    #: The entity representations
+    entity_representations: Sequence[NewRepresentationModule]
+
+    #: The relation representations
+    relation_representations: Sequence[NewRepresentationModule]
+
+    #: The weight regularizers
+    weight_regularizers: List[Regularizer]
+
+    def __init__(
+        self,
+        triples_factory: TriplesFactory,
+        interaction: Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation],
+        entity_representations: EmbeddingSpecificationHint = None,
+        relation_representations: EmbeddingSpecificationHint = None,
+        loss: Optional[Loss] = None,
+        predict_with_sigmoid: bool = False,
+        preferred_device: DeviceHint = None,
+        random_seed: Optional[int] = None,
+    ) -> None:
+        """Initialize the module.
+
+        :param triples_factory:
+            The triples factory facilitates access to the dataset.
+        :param loss:
```

Comment:
missing parameters: interaction, entity_representations, relation_representations

Reply:
Done in 50245c60 (also added darglint)